### ISSUE TITLE
Grappler Errors with ConvLSTM2D

### ISSUE THREAD

**Issue by acfrock** [keras-team/keras#16754]
### System information

-   **Have I written custom code (as opposed to using a stock example script
    provided in TensorFlow)**: No
-   **OS Platform and Distribution**: Windows 11 Build 22000
-   **TensorFlow installed from**: pip package
-   **TensorFlow version**: v2.9.0-18-gd8ce9f9c301 2.9.1
-   **Python version**: Python 3.10.4
-   **CUDA/cuDNN version**: CUDA 11.7.0-1
-   **GPU model and memory**: Nvidia 2080 Ti 11GiB Memory
-   **Exact command to reproduce**: `python conv_lstm.py` with [conv_lstm.py](https://github.com/keras-team/keras-io/blob/master/examples/vision/conv_lstm.py) from the official Keras examples

### Describe the problem

Using the ConvLSTM2D layer produces a grappler error: `layout failed: INVALID_ARGUMENT: MutableGraphView::SortTopologically error: detected edge(s) creating cycle(s) ...` when run with a GPU. Interestingly, this does not happen if TensorFlow is forced to run without a GPU.

It is not clear whether this is a problem with grappler, with the implementation of the ConvLSTM2D layer, or a bug in CUDA.

### Contributing
- Do you want to contribute a PR?: no

### Source code / logs
I used the example script from the Keras documentation. [[Keras example page](https://keras.io/examples/vision/conv_lstm/)] [[script](https://github.com/keras-team/keras-io/blob/master/examples/vision/conv_lstm.py)]. The logs are here: 
[output.txt](https://github.com/keras-team/keras/files/9039413/output.txt).

**Comment by sushreebarsa** [comment]
@acfrock I tried to run the example script for replicating the issue and didn’t face the error reported. Could you find the gist [here](https://colab.research.google.com/gist/sushreebarsa/917d8ee540e79e34a57e091360bd45a3/conv_lstm.ipynb#scrollTo=N4p596xMrKhY) and let me know if I am missing something to reproduce the issue? 
Thank you!

**Comment by acfrock** [comment]
The error message comes from the CPP logger, so by default it doesn't show in Colab. By using wurlitzer we can see the message: [gist](https://colab.research.google.com/gist/acfrock/1747352648b82cfccd0b18d20be41f70/conv_lstm.ipynb#scrollTo=lulIT_Waif3a).

**Comment by sushreebarsa** [comment]
@acfrock Thank you for the update!
@gadagashwini Please find the reproducible gist [here](https://colab.research.google.com/gist/sushreebarsa/190125ccf11e1316ae88489ebd2f2a9b/conv_lstm.ipynb) of the reported issue for your reference. Thank you!

**Comment by gowthamkpr** [comment]
@acfrock This is a grappler error. Please take a look at the workaround provided [here](https://github.com/tensorflow/tensorflow/issues/36901#issuecomment-696398497). Also, please post this issue [here](https://github.com/tensorflow/tensorflow/issues/new?assignees=&labels=&template=tensorflow_issue_template.yaml). Thanks!

**Comment by t-kalinowski** [comment]
I've run into the same problem and am subscribed to the issue. Posting to keep the stale bot away.

**Comment by gowthamkpr** [comment]
@acfrock @t-kalinowski Looks like this issue is related to this [issue](https://github.com/tensorflow/tensorflow/issues/36901). There has been a [PR](https://github.com/keras-team/keras/commit/214d11055890d5f3dcc83d2af69ad17089b8177d) that has been merged. Please confirm if otherwise. Thanks!
